app/controllers/homepage_controller.rb

```diff
@@ -204,6 +208,23 @@ def find_listings(params:, current_page:, listings_per_page:, filter_params:, in
     end
   end
 
+  # Time to cache category translations per locale
+  CATEGORY_DISPLAY_NAME_CACHE_EXPIRE_TIME = 24.hours
```

Comment:
@bladealslayer 
For such a long period of time Category model must have a hook after_save cleans up the cache for this community category names. Admin changes in the categories must be available immediately in front-end.

Reply:
The cache key is formed based on top categories and their modification time. Child categories and translations all have `touch: true` for "parent", so whenever anything is changed, the cache is effectively invalidated. Seems to work fine in my testing.